Windows redefinition of struct timeval

MinGW with msys:

```
CC    mrbgems/mruby-time/src/time.c -> build/host/mrbgems/mruby-time/src/time.o
c:/Users/Mitchell/code/3rdparty/mruby/mrbgems/mruby-time/src/time.c:56:8: error: redefinition of 'struct timeval'
 struct timeval {
        ^
In file included from c:/MinGW/mingw64/x86_64-w64-mingw32/include/time.h:256:0,
                 from c:/Users/Mitchell/code/3rdparty/mruby/mrbgems/mruby-time/src/time.c:8:
c:/MinGW/mingw64/x86_64-w64-mingw32/include/_timeval.h:10:8: note: originally defined here
 struct timeval
        ^
```

--- a/mrbgems/mruby-time/src/time.c
+++ b/mrbgems/mruby-time/src/time.c
@@ -53,10 +53,14 @@
 #  endif
 
 typedef long suseconds_t;
+
+# ifndef __MINGW64__
 struct timeval {
   time_t tv_sec;
   suseconds_t tv_usec;
 };
+# endif
+
 static int
 gettimeofday(struct timeval *tv, void *tz)
 {

